### ISSUE TITLE
Mouse lag/stutter whenever using mumble

### ISSUE THREAD

**Issue by ryanjohndunne** [mumble-voip/mumble#3275]
Whenever I use mumble I notice some mouse lag/stuttering. 
It's most noticeable when I first open the program but every now and then it may randomly freeze the mouse.
The 64 bit snapshot builds cause much worse stuttering than the 32 bit client. You will see in the video below that the mouse randomly jumps across the screen. When using the 64 bit snapshot the mouse will only jump, it won't even make a laggy movement to where I'm pointing it - it just freezes in spot and jumps randomly across the screen.

Here is a video of the problem: https://youtu.be/7RTlBL0HNaA
Do note that I was attempting to draw the circle at the same speed as I opened mumble (you can see it gets a lot slower, with some lag, and then randomly jumps halfway across the screen). This problem can randomly occur whenever mumble is open but always happens when first opening mumble. This problem never happens when mumble is closed - tried with discord and team speak, both don't have this issue.

Computer Specs:
CPU: 6700k @ 4.7ghz (problem still happens when stock speed)
GPU: Nvidia 1080ti
RAM: 16gb memory @ 3200hz (problem still happens with lower ram speed)
STORAGE: 512gb Samsung evo pro with 1tb western digital black HDD (windows and mumble installed on the samsung)
MOTHERBOARD: MSI gaming m7 zi70a
MOUSE: G502 spectrum @ 1000hz polling rate. I have noticed that lower polling rate on the mouse can reduce the lag when using mumble. But I should not have to lower my mouses polling rate to fix this issue. I also tried this on an original g502 (non spectrum version) and it has the same mouse lag.
MONITORS: Acer predator xb271hu 2560x1440p @ 144hz with a secondary asus vg248qe 1920x1080 @ 144hz. The mouse lags on both monitors when mumble is opened.
Operating system: windows 10.
All of my drivers are up to date.

**Comment by mkrautz** [comment]
Thanks for the detailed report.

So far, I don't know any of us developers being able to reproduce this.

I assume you've got Logitech Gaming Software installed? (I don't, so maybe that's a difference.)

**Comment by assbutt69** [comment]
I have a [similar issue](https://github.com/mumble-voip/mumble/issues/3271) and I'm also using a Logitech mouse. I don't have Logitech Gaming Software installed. With their newer mice you just install the drivers once to change the settings and then they're saved on the mouse's on board memory, at which point you can uninstall the drivers.

**Comment by davidebeatrici** [comment]
@ryanjohndunne Can you check your mouse's polling rate as @assbutt69 did in #3271?

**Comment by ryanjohndunne** [comment]
@mkrautz It happens when I have the software installed or uninstalled, You're able to load the profile onto the mouses on-board memory and use that rather than use the software. I reformatted multiple times trying to fix this problem and even had just mumble installed - still happens.

@davidebeatrici Yeah as I mentioned at the bottom of the first post lower polling rate does reduce the stuttering but it doesn't completely solve it. I have to put it down from 1000 to 250 to see some improvement. 

The weird thing is I have used the Logitech g502 for many years and this issue only started happening maybe 6 months ago. I think it has something to do with a combination of windows 10 with the newest updates + Logitech mouse + mumble.

I used to be able to trigger the stuttering when changing my sound from my headphones to my speakers through the windows 10 settings and installing new software sometimes triggers it. Installing new GPU drivers almost always freezes the mouse too, very strange. I initially thought it had something to do with the GPU trying to play mumble through the speakers when the headset was connected and causing some error - as I use audio over display port and play through my monitors speakers some times. However, it only causes mouse stuttering when I change the sound output in windows 10 on the 64 bit snapshot and not when I use the 32 bit client, This all only happens when mumble is open.

**Comment by Zyl9393** [comment]
I am having this problem ever since installing FACEIT Anti-Cheat. It did not happen before that. Probably some hyper obscure code injections are the cause.

**Comment by assbutt69** [comment]
I recently upgraded to a new video card (went from an R7 260x to a GTX960, and I started getting weird stuttering in games like CSGO and R6: Siege. Essentially whenever the fps would dip a tiny bit (like from 120 to 110 for instance) my mouse inputs would start dropping in a manner that feels like I lifted the mouse too far off of the mousepad.

I did open an issue a while ago (in 2017) where I mentioned that having Mumble open makes my polling rate unstable, which is still the case, only now I've tested with Windows Hooks and XInput turned off in the shortcuts tab, which makes no difference. What I just figured out today is that it's Mumble that's causing these stutters for me. In addition to the polling rate going back to normal while Mumble isn't open, I also don't experience any stuttering at all.

To summarize:
From 2017 until recently, when I was using an R7 260x: mouse polling rate would be slightly unstable while Mumble is open, but I didn't experience any dropping of mouse input

From a couple months ago until now, when I upgraded to a GTX 960: while Mumble is open the unstable polling rate issue still happens, but my mouse stutters/drops inputs in games that aren't running >= my refresh rate (120) at all times as well

**Comment by Jektt** [comment]
After troubleshooting why my PC was having massive slowdowns, I've narrowed it down to mumble as @ryanjohndunne did. 

64 bit version of Mumble 1.3 causes essentially a complete system lock for a few seconds. The mouse drags across the screen and the entire OS is unresponsive until it recovers. As ryanjohndunne mentioned, disabling and enabling sound devices allows this to be reproduced consistently, although it occurs seemingly at random when this isn't forced. 

I also have a Logitech mouse, but having the Logitech Gaming Software installed didn't impact the stutters. I have a Vega 64, thought it was a GPU driver issue but after downgrading to a 6 month old driver I realised the problem was still happening.

Installing the 32 bit version of 1.3 fixes the problem. There is still a compartively minor stutter when it starts for the first time, but after that no problems and disabling/enabling sound devices has no effect.

The 64 bit version of Mumble can actually be running and you can enable/disable sound devices without the stutter happening IF the microphone privacy settings are set to prevent apps from accessing the microphone. That may be useful.

**Comment by davidebeatrici** [comment]
Closing in favor of #4039.
